gvm-libs fails to build since gvm-libs >= 22.5.2

### Expected behavior
gvm-libs >= 22.5.2 building without running into compile failure. building gvm-libs 22.5.1 still compiles fine so it's probably due to https://github.com/greenbone/gvm-libs/commit/4557386c.

### Actual behavior
Run into failure to build from source.

### Steps to reproduce
1. have glib >= 2.76.2 installed
2. try to build gvm-libs >= 22.5.2
3. run into error

```
[ 15%] Building C object base/CMakeFiles/gvm_base_shared.dir/networking.c.o
cd /var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/build/base && /usr/bin/x86_64-pc-linux-gnu-cc -DGVM_LIBS_VERSION=\"22.5.2\" -DGVM_SYSCONF_DIR=\"/etc/gvm\" -DREDIS_SOCKET_PATH=\"/tmp/redis.sock\" -Dgvm_base_shared_EXPORTS -I/usr/x86_64-pc-linux-gnu/include/glib-2.0 -I/usr/x86_64-pc-linux-gnu/lib/glib-2.0/include -pipe -O2 -march=native      -std=c11     -Wall     -Wextra     -Werror     -Wpedantic     -Wmissing-prototypes     -Wshadow     -Wsequence-point     -D_ISOC11_SOURCE     -D_DEFAULT_SOURCE -O3 -DNDEBUG -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector -std=gnu11 -fPIC -MD -MT base/CMakeFiles/gvm_base_shared.dir/networking.c.o -MF CMakeFiles/gvm_base_shared.dir/networking.c.o.d -o CMakeFiles/gvm_base_shared.dir/networking.c.o -c /var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/gvm-libs-22.5.2/base/networking.c
/var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/gvm-libs-22.5.2/base/networking.c: In function 'gvm_resolve_list':
/var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/gvm-libs-22.5.2/base/networking.c:378:11: error: 'g_memdup' is deprecated: Use 'g_memdup2' instead [-Werror=deprecated-declarations]
  378 |           list = g_slist_prepend (list, memdup (&dst, sizeof (dst)));
      |           ^~~~
In file included from /usr/x86_64-pc-linux-gnu/include/glib-2.0/glib/gstring.h:37,
                 from /usr/x86_64-pc-linux-gnu/include/glib-2.0/glib/giochannel.h:36,
                 from /usr/x86_64-pc-linux-gnu/include/glib-2.0/glib.h:56,
                 from /var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/gvm-libs-22.5.2/base/array.h:28,
                 from /var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/gvm-libs-22.5.2/base/networking.h:28,
                 from /var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/gvm-libs-22.5.2/base/networking.c:25:
/usr/x86_64-pc-linux-gnu/include/glib-2.0/glib/gstrfuncs.h:350:23: note: declared here
  350 | gpointer              g_memdup         (gconstpointer mem,
      |                       ^~~~~~~~
/var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/gvm-libs-22.5.2/base/networking.c:384:11: error: 'g_memdup' is deprecated: Use 'g_memdup2' instead [-Werror=deprecated-declarations]
  384 |           list = g_slist_prepend (list, memdup (&dst, sizeof (dst)));
      |           ^~~~
/usr/x86_64-pc-linux-gnu/include/glib-2.0/glib/gstrfuncs.h:350:23: note: declared here
  350 | gpointer              g_memdup         (gconstpointer mem,
      |                       ^~~~~~~~

Error:
  * In program cave perform install --hooks --managed-output --output-exclusivity with-others =net-analyzer/gvm-libs-22.5.2:0::net --destination installed --replacing =net-analyzer/gvm-libs-22.5.1:0::installed --x-of-y 1 of 1:
  * When installing 'net-analyzer/gvm-libs-22.5.2:0::net' replacing { 'net-analyzer/gvm-libs-22.5.1:0::installed' }:
  * When running an ebuild command on 'net-analyzer/gvm-libs-22.5.2:0::net':
  * Install failed for 'net-analyzer/gvm-libs-22.5.2:0::net' (paludis::ActionFailedError)

make[2]: Leaving directory '/var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/build'
make[1]: Leaving directory '/var/tmp/paludis/build/net-analyzer-gvm-libs-22.5.2/work/build'
cc1: all warnings being treated as errors
make[2]: *** [base/CMakeFiles/gvm_base_shared.dir/build.make:163: base/CMakeFiles/gvm_base_shared.dir/networking.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:198: base/CMakeFiles/gvm_base_shared.dir/all] Error 2
make: *** [Makefile:159: all] Error 2
```

### GVM versions
**gsa:** Greenbone Security Assistant 22.04.1
**gvm:** Greenbone Vulnerability Manager 22.4.2
**openvas:** OpenVAS 22.5.0
**gvm-libs:** gvm-libs 22.5.1

### Environment
**Operating system:** Exherbo Linux
**Installation method / source:** source-based package installation

### Logfiles
[gvm-libs-22.5.2.buildlog.txt](https://github.com/greenbone/gvm-libs/files/11485536/gvm-libs-22.5.2.buildlog.txt)

--- a/base/networking.c
+++ b/base/networking.c
@@ -48,8 +48,7 @@
  */
 #define G_LOG_DOMAIN "libgvm base"
 
-#if (GLIB_MAJOR_VERSION >= 2) && (GLIB_MINOR_VERSION >= 67) \
-  && (GLIB_MICRO_VERSION >= 3)
+#if GLIB_CHECK_VERSION(2, 67, 3)
 #define memdup g_memdup2
 #else
 #define memdup g_memdup

--- a/util/kb.c
+++ b/util/kb.c
@@ -40,8 +40,7 @@
  */
 #define G_LOG_DOMAIN "libgvm util"
 
-#if (GLIB_MAJOR_VERSION >= 2) && (GLIB_MINOR_VERSION >= 67) \
-  && (GLIB_MICRO_VERSION >= 3)
+#if GLIB_CHECK_VERSION(2, 67, 3)
 #define memdup g_memdup2
 #else
 #define memdup g_memdup

